### PR TITLE
Several Changes to Configuration

### PR DESCRIPTION
**Pull request by haz**
- [x] Environment varible for frontend port
- [x] Unified `requirements.txt`
- [x] Updated packages
- [x] Fixed bugs + testing with all the services

### CODE REVIEW

**Comment by haz** [comment]
Alright, set for some testing, @nirlipo . Appears to work with a sudo-less make, and I've confirmed at least lama-first returns a plan, and can see all the details via the monitor (5555).

**Comment by haz** [comment]
Closes #57 and #61

**Comment by haz** [comment]
Able to have a look, @nirlipo ?

File: server/Dockerfile [hunk 1]
```diff
@@ -3,24 +3,34 @@
...
 # Satisficing classical planning
 RUN planutils install -f -y dual-bfws-ffparser
 
 #Optimal classical planning
 RUN planutils install -f -y delfi
 
-# top-k classical planning
-RUN planutils install -f -y forbiditerative-topk
+# top-k classical planning -- currently unavailable
+# RUN planutils install -f -y forbiditerative-topk
```
Comment (nirlipo) [inline server/Dockerfile:15]: Is there an issue we should track from planutils pointing to what's wrong with the top-k planner?

File: server/docker-compose.yml [hunk 2]
```diff
@@ -7,7 +6,7 @@ services:
       dockerfile: Dockerfile
     restart: always
     ports:
-     - "80:80"
+    - "${FRONTEND_PORT:-80}:80"
```
Comment (nirlipo) [inline server/docker-compose.yml:9]: Is this variable defined somewhere else?

**Comment by nirlipo** [comment]
One reason I need to run make with sudo is to access the private key used to set up HTTPS. I'm not sure it would be a good idea to change the key's access rights. If someone is deploying it with http, it should run without sudo.

@haz I also manually added the nginx service in our docker-compose file at the server so we could get our landing page in https://solver.planning.domains/

I had to upgrade our docker compose package, but that's not a story for this thread :)

For this to work, I had to change the port - "${FRONTEND_PORT:-80}:80" to port 3000 in the docker-compose file and the frontend/Dockerfile. Let me know if there's an intended behaviour I'm missing here.

If the intention is to have the frontend working with nginx, I'll push and https, I'll do another pull request

**Comment by nirlipo** [comment]
The server is running our usual, Flower is working with a shiny new interface, and top-k is missing from the dropdown menu as it has been removed from the server.

File: server/docker-compose.yml (hunk 2 above)
Comment (haz) [reply server/docker-compose.yml:9]: In the sample environment file now, ya.

**Comment by haz** [comment]
Ah, right, the https needing root (for access to those keys) makes sense.

Do we really need nginx? I was able to get things working locally without it. The port for the frontend should be defined in the `.env` file.

**Comment by nirlipo** [comment]
I don't get served the current page you can see in https://solver.planning.domains/ unless I run nginx.

**Comment by haz** [comment]
Yep, this makes total sense now that I look at the PaaS config. We could either run nginx natively, or pull it into the docker-compose like you have it, but either approx is likely out-of-scope for PaaS, right? Very specific to the individual deployments, and only necessary if the frontend is needed at a specific URL.

**Comment by nirlipo** [comment]
Cool. Let me revise how we deploy the front-end. It may be best to have it even completely separated from the current repo. This will make it easier to merge future changes.

**Pull request merged**